### DISCUSSION TITLE
Get resolver from useForm return, and schema from resolver

### DISCUSSION THREAD

**Discussion by appelsinvandi** [react-hook-form/react-hook-form#6812] [Ideas]
As far as I can tell, there's currently no way to retrieve the resolver from a `useForm` return, and likewise no way to retrieve a schema from a `resolver` return.

I use `zod` for schema validation, and would love to more easily be able to use zod's introspection methods to make inputs that already use `useFormContext` to determine certain states, be even smarter.

Such a component could potentially look like this for zod:

```jsx
import { get, useFormContext } from 'react-hook-form'

const HookedInput = ({ name }) => {
  const { register, resolver } = useFormContext()

  // The below line would return the ZodObject schema
  const schema = resolver.schema
  const schemaEntry = get(schema.shape, name) 

  return (
    <input type="text" {...register(name)} required={!schemaEntry.isOptional()} />
  )
}
```

As I believe you can't edit a resolver's schema, and have to pass an entirely new resolver to change it, it should also force a re-render to any component using to it, ensuring that components will still be updated if it changes.

This can of course already be done, by just creating a separate context for the schema yourself. But I thought I would put the idea out there, as I believe it would be convenient to have it all in one place.

I couldn't find any similar discussion or issue, searching for "get resolver" and "retrieve resolver".

**Comment by bluebill1049** [comment]
i belive this is related feature: https://github.com/react-hook-form/react-hook-form/discussions/6472

**Comment by estevaolucas** [comment]
@appelsinvandi have you found any workaround for this limitation? I need to access resolvers within my field, but didn't want to create a custom context just for that.

**Reply by appelsinvandi** [reply]
Sadly no. I gave it a good shot back then, but ultimately had to create a custom context wrapping around the default one with the extra features I wanted. I guess it's not such a big deal, it just felt a little "hacky."

**Reply by estevaolucas** [reply]
yeah, I was trying to avoid this. 

I've noticed `control` holds an internal `_option.resolver` and `_executeSchema`. Not sure if I get if a field is required or not from there.

**Reply by martinratinaud** [reply]
I checked and found no way of getting back the schema

Using `_option.resolver` and `_executeSchema`, you can get the validated values in a promise :-( but that's it

```js
const context = useFormContext();

  React.useEffect(() => {
    (async () => {
      console.log(context.control);//eslint-disable-line
      console.log(await context.control._executeSchema([field.name]));//eslint-disable-line
      console.log(await context.control._options.resolver({[field.name]: field.value}, context, {}));//eslint-disable-line
    })();
  }, []);
```

**Reply by SaugatGhimire321** [reply]
It seems there's no way of getting the schema.

Only option is to pass the schema to the context.

**Reply by jaaymes** [reply]
@SaugatGhimire321 

try this

`  React.useEffect(() => {
    (async () => {
      console.log(await control._runSchema([name]));
    })();
  }, [control, name]);`
